clang/lib/Sema/SemaCUDA.cpp

```diff
@@ -1000,13 +1000,9 @@ void Sema::checkCUDATargetOverload(FunctionDecl *NewFD,
     // should have the same implementation on both sides.
     if (NewTarget != OldTarget &&
         ((NewTarget == CFT_HostDevice &&
-          !(LangOpts.OffloadImplicitHostDeviceTemplates &&
-            isCUDAImplicitHostDeviceFunction(NewFD) &&
-            OldTarget == CFT_Device)) ||
+          !isCUDAImplicitHostDeviceFunction(NewFD)) ||
```

Comment:
This change was previously excluding device functions. Do we want/need to keep that check in place?
If it does need to be gone, then the comment above may need to be updated.

Reply:
actually we do not need this part of change to fix this issue, since this part only affects overloading. As we fixed the overloading/overriding differentiation, this part is not needed. I have removed this part in the updated change.